jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/DataTest.java

```diff
@@ -329,6 +329,11 @@ public void testDataInputAssociationsWithPojoPartial() throws Exception {
         internalTestDataInputAssociationWithPojo("BPMN2-DataInputAssociations-Pojo.bpmn2");
     }
 
+    @Test
+    public void testDataInputAssociationsWithPojoWithoutsource() throws Exception {
```

Comment:
I think it would be good to add another similar test case to check dataInputsAssociations work when only `sourceRef` is defined.
Same for dataOutputAssociations.

Reply:
These scenarios are covered by testDataInputAssociationsStunnerWithPojo and testDataOutputAssociationsStunnerWithPojo

Reply:
I can't find these scenarios in the test cases you mention. 

1. For dataInputAssociations I don't see any association having only `sourceRef` element. All of them are coming with `targetRef`
2. For dataOutputAssociations I don't see any association having only `targetRef` element. There is only one with `sourceRef` only.

https://github.com/kiegroup/jbpm/blob/f805920c306411ae6385357cc81e0045af272a0b/jbpm-bpmn2/src/test/resources/BPMN2-DataInputAssociationStunner.bpmn2

https://github.com/kiegroup/jbpm/blob/f805920c306411ae6385357cc81e0045af272a0b/jbpm-bpmn2/src/test/resources/BPMN2-DataOutputAssociationStunner.bpmn2

Reply:
Ok, I misunderstood your comment. You are right, there are not, but it is ok, since stunner only supports the ones that are tested. Think that this change "breaks" BPMN assumption that both sourceRef and targetRef are going to be present for assignments and we are just covering the cases where stunner does not obey that rule